sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3582,6 +3582,163 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     checkAnswer(sql("SELECT 0 FROM ( SELECT * FROM B JOIN C USING (id)) " +
       "JOIN ( SELECT * FROM B JOIN C USING (id)) USING (id)"), Row(0))
   }
+
+  test("test casts pushdown on orc/parquet for integral types") {
```

Comment:
Just a tip. In general, Apache Spark community recommends not to add to `SQLQuerySuite` because this file is already fat. 
- For general SQL operation, we prefer SQL file test.
- For data source only test, we prefer to use more smaller suite. For your test case,  `FileBasedDataSourceSuite` might be better. Please take a look if that is proper to you.

Reply:
Okay. I think `FileBasedDataSourceSuite` may be a better place. Let me move these to there.